controller/users.go

```diff
@@ -157,6 +193,137 @@ func (c *UsersController) getKeycloakProfileInformation(ctx context.Context, tok
 	return response, err
 }
 
+func (c *UsersController) updateUserInDB(id *uuid.UUID, ctx *app.UpdateUserAsServiceAccountUsersContext) error {
+
+	// We'll refactor the old Users update API to be redirected to Auth service in the near future.
+	// Hence, not spending time in refactoring that to consume this function.
+
+	returnResponse := application.Transactional(c.db, func(appl application.Application) error {
+		identity, err := appl.Identities().Load(ctx, *id)
+		if err != nil || identity == nil {
+			log.Error(ctx, map[string]interface{}{
+				"identity_id": id,
+			}, "auth token contains id %s of unknown Identity", *id)
+			jerrors, _ := jsonapi.ErrorToJSONAPIErrors(ctx, goa.ErrUnauthorized(fmt.Sprintf("Auth token contains id %s of unknown Identity\n", *id)))
```

Comment:
As said above it's not an error if the identity doesn't exist. Just create a new one. And the error message is not quite correct. We don't pass the ID in the token.

Reply:
https://github.com/fabric8-services/fabric8-wit/pull/1637/commits/ba25ff94d34ee78bf102e6bb33505105bec18344